### PR TITLE
Tmi2 417 send batch to spotlight

### PR DESCRIPTION
**Pull request by GavCookCO**
## Description

fetches spotlight batches from the database, converts the data into a spotlight appropriate format and sends to spotlight for processing. 

Ticket # and link

https://technologyprogramme.atlassian.net/jira/software/projects/TMI2/boards/331?selectedIssue=TMI2-417

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

File: src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
+
+    private final ObjectMapper jacksonObjectMapper;
+
+    private final RestTemplate restTemplate;
+
     public boolean existsByStatusAndMaxBatchSize(SpotlightBatchStatus status, int maxSize) {
         return spotlightBatchRepository.existsByStatusAndSpotlightSubmissionsSizeLessThan(status, maxSize);
     }
 
+    // TODO refactor this - it can potentially return more than one result and will cause
```
Comment (dylanwrightCO) [inline src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java:57]: Is this getting actioned as part of another ticket?

File: src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java [hunk 2]
```diff
@@ -51,4 +85,136 @@ private SpotlightBatch getSpotlightBatch(UUID spotlightBatchId) {
...
+    public List<SpotlightBatch> getSpotlightBatchesByStatus(SpotlightBatchStatus status) {
+        return spotlightBatchRepository.findByStatus(status).orElse(new ArrayList<>());
+    }
+
+    public List<SendToSpotlightDto> generateSendToSpotlightDtosList(SpotlightBatchStatus status) {
+
+        final List<SendToSpotlightDto> sendToSpotlightDtos = new ArrayList<>();
+        final List<SpotlightBatch> spotlightBatches = getSpotlightBatchesByStatus(status);
+
+        // gav comment - think we could turn this loop into a stream to make the method
```
Comment (dylanwrightCO) [inline src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java:97]: Same here, is this getting actioned as part of another ticket?

Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightBatchService.java:97]: I'll remove the comment and we can refactor in future. Don't think we'll have enough time just now

**Review by dylanwrightCO — APPROVED** [review]

**Comment by dylanwrightCO** [comment]
Approving on basis that a new ticket is created to capture comments.

**Pull request merged**